### ISSUE TITLE
Traits

### ISSUE THREAD

**Issue by DanielWillett** [UncreatedStaff/UncreatedWarfare#40]
Would be a temporary status effect bought with credits.
Able to be blacklisted or whitelisted for gamemodes
Able to be blacklisted or whitelisted for kit classes
Examples:
  self revive kit,
  more resistance / armor
  better stamina consumption
  basically virtual items that would help you for a temporary amount of time

**Comment by DanielWillett** [comment]
This would be a possible fix for issue #31

**Comment by HamzaXSniper** [comment]
It sounds like a great idea perhaps having the rank warrant officer would be needed to purchase/activate those powerups But it might seem too complicated or too modded for some players thinking its like cheating or overpowered. New players also might consider it "pay 2 win" and discourage them from playing.

**Comment by 3312bKarp** [comment]
I dont think grindable powerups would either fit or be fair to play against

**Comment by DanielWillett** [comment]
Yeah, i didn't really want to call them powerups cause it sounds like something out of fortnite but couldn't think of a better word. They would be more play-to-win than pay-to-win, but right now credits are almost completely useless once you've unlocked all the kits.

**Comment by HamzaXSniper** [comment]
I agree with what karp said; imagine you killing an entire squad and then a medic self revives like its apex and kills you before reviving his entire squad.

**Comment by benvalkin** [comment]
I kind of like the idea of perks/attributes that either encourage teamwork-based benefits or are very situational for a short amount of time. They should avoid directly improving players' health, damage ability, running skills, ammo, etc. They can be displayed and purchased from signs, and we can also give them funny names like in Call of Duty.
Some ideas:

- 'Motivated' - Shovel as fast as a combat engineer
- 'Intimidation' - Your squad can swing a contested flag in your team's favor even when outnumbered
- 'Bad Omen' - ominous notification when enemy mortars are about to land
- 'Skilled Crewman' - everyone in your armored vehicle gets extra XP and credits for kills/assists
- 'Strength in Numbers' - Your squad gets +10% armor when fighting near each other
- 'Ghost' -Become invisible UAV when on foot
- 'Guided By God' - Heat-seaking missiles fired from ground AA are 15% more responsive
- 'Superheated' - Countermeasure flares are more effective at distracting missiles

There's also always physical items too:
- Self revive (expensive)
- Special fortifications
- Sentry Gun - expensive (and probably not very powerful) sentry gun that you can build

**Comment by SunShine214** [comment]
I like this. The self revive would be very nice. Maybe a power up should be "Commanding" where people in your squad can be forced to rally. Probably a terrible idea, but people in squads rarely cooperate

**Comment by HamzaXSniper** [comment]
I can see that being abused alot.

**Comment by DanielWillett** [comment]
I do like those that's sort of the kind of perks I was thinking as well, just couldn't really think of any ideas at the time.

**Comment by DMJaxun** [comment]
I'd rather not become EA games

**Comment by DanielWillett** [comment]
They would be for credits not actual money.

**Comment by DanielWillett** [comment]
Would like to think of a better term for them however.

**Comment by DanielWillett** [comment]
Still also need to add a /trait command to edit trait properties. Low priority.

**Comment by DanielWillett** [comment]
Traits will not activate until after the staging phase has started (your timer doesn't tick down until the game starts).
Lots of testing and fixes were done for traits.
New trait command for giving and revoking traits and setting properties from in-game.
Signs look nicer and are more responsive.

Still required: implementing Superheated and GuidedByGod into the guided missile code.

**Comment by DanielWillett** [comment]
Also need to make sure 6 buffs aren't already queued when adding a new one during staging phase...

**Comment by DanielWillett** [comment]
Add max xp trait
- Hamza

**Comment by HamzaXSniper** [comment]
Actually its max skill trait🤓
